pkg/ccl/backupccl/data_bundle.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package backupccl
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+)
+
+// dataBundle specifies the data that is to be restored in a restoration flow.
+// It can specify spans to be restored, as well as system tables that should be
+// restored in the case of cluster backups.
+// It also includes peripheral data, related to the data that is being restored
+// such as rekey information and PK IDs to count the ingested rows.
+type dataBundle interface {
```

Comment:
naming nit (the best kind): I'm not sure "dataBundle" really makes me understand what it is, but I'm not sure what would? restoreSpec? restorationTask? I donno. not blocking since could just as easily revisit later if we think of something else.